11-Backbonejs/Backbone-Views-of-Collections.md

```diff
@@ -0,0 +1,195 @@
+import $ from 'jquery';
+import TaskView from '../views/task_view';
+
+
+const TaskListView = Backbone.View.extend({
+  initialize: function(params) {
+    this.template = params.template;
+  },
+  render: function() {
+    // Clear the unordered list
+    this.$('#todo-items').empty();
+    // Iterate through the list rendering each Task
+    this.model.each((task) => {
+      // Create a new TaskView with the model & template
+      const taskView = new TaskView({
+        model: task,
+        template: this.template,
+        tagName: 'li',
+        className: 'task',
+      });
+      // Then render the TaskView
+      // And append the resulting HTML to the DOM.
+      this.$('#todo-items').append(taskView.render().$el);
+    });
+    return this;
+  }
+});
+
+export default TaskListView;
+```
+Just like in `TaskView`, in `initialize` we store the template in an attribute of the view.  The view's render method looks like the render method we wrote earlier with a few differences.
+
+We finish the `render` method by appending each TaskView to the list inside the DOM and when the loop is finished we `return this;` by convention.  It's a fairly common pattern developers use in JavaScript to deal with confusing `this` issues.
+
+
+### About `this.$`
+
+In the `render` method we clear the list using `this.$`.  The method `this.$` performs a jQuery selection for HTML inside `el`.  Using `this.$` we can ensure that our view does not interact with any code outside of it.  You should never use jQuery directly within a view.  Instead you should use `this.$` just as you would jQuery to avoid impacting the DOM outside your view.  
+
+
+### Updating `app.js`
+
+We will adjust our `$(document).ready` handler to use the `TaskListView`.
+
+
+```javascript
+// app.js
+// imports etc...
+
+import TaskListView from './views/task_list_view';
+
+// ...
+
+$(document).ready(function() {
+  // Why does this belong in $(document).ready ?
+  const taskListView = new TaskListView({
+    model: taskList,
+    template: _.template($('#task-template').html()),
+    el: 'main',
+  });
+
+  taskListView.render();
+});
+```
+
+We create the TaskListView and set it's model to be our taskList collection and set the template attribute.  In this case we start off our View attached to an element in the DOM, in this case our `main` tag.
+
+Lastly, we call render on the taskListView.
+
+Let's take the time to also remove the references to `renderList` and `TaskView` in `app.js`, as they are now unused!
+
+### Handling Creating New Tasks
+
+Just like we added event handlers in the `TaskView` to handle button clicks, we can add an event handler to create a new task and a method to read from the new task form, and add them to the collection. Let's move our logic to add a new task into our view.
+
+![Creating New Tasks](images/addButton.png)
+
+
+```javascript
+// views/task_list_view.js
+import Task from '../models/task'
+
+...
+}, // end of render
+
+  events: {
```

Comment:
This is a huge amount of code in one step.  Breaking this into sections would be better I think.